geomesa-memory/src/main/scala/org/locationtech/geomesa/memory/cqengine/utils/SFTAttributes.scala

```diff
@@ -0,0 +1,59 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.memory.cqengine.utils
+
+import java.util.UUID
+
+import com.googlecode.cqengine.attribute.Attribute
+import com.vividsolutions.jts.geom.Geometry
+import org.locationtech.geomesa.memory.cqengine.attribute.{SimpleFeatureFidAttribute, SimpleFeatureAttribute}
+import org.opengis.feature.`type`.AttributeDescriptor
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConversions._
+
+case class SFTAttributes(sft: SimpleFeatureType) {
+  private val attributes = sft.getAttributeDescriptors
+
+  private val lookupMap: Map[String, Attribute[SimpleFeature, _]] = attributes.map { attr =>
+    val name = attr.getLocalName
+    name -> SFTAttributes.buildSimpleFeatureAttribute(attr.getType.getBinding, name)
+  }.toMap
+
+  // TODO: this is really, really bad :)
+  def lookup[T](attributeName: String): Attribute[SimpleFeature, T] = {
```

Comment:
Obvious, no. Did add a followup ticket in case we want to revisit: https://geomesa.atlassian.net/browse/GEOMESA-1408